Adopt UniqueRef in JavaScriptCore/inspector

annevk opened the pull request:
#### 9de4e9c32d93dab9d7f4e68f42575c2700cd7a7e
<pre>
Adopt UniqueRef in JavaScriptCore/inspector
<a href="https://bugs.webkit.org/show_bug.cgi?id=295452">https://bugs.webkit.org/show_bug.cgi?id=295452</a>

Reviewed by Youenn Fablet.

Also mark some additional member variables const and make inspector
code more consistent throughout.

Helps with <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines</a>

Canonical link: <a href="https://commits.webkit.org/297041@main">https://commits.webkit.org/297041@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/86957b7042e52c1f31c0b72ce9b33c0f15b10fd5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110318 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/29977 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20409 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116342 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60567 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112281 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30656 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38564 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83880 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113266 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24458 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99341 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64326 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23818 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17480 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60138 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/102809 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93838 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17536 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119132 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/108872 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37358 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27706 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92851 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37730 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95611 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92677 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37675 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15407 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33252 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17804 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37252 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42723 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/133147 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36914 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35989 "Found 2 new JSC stress test failures: wasm.yaml/wasm/stress/jitless-wasm-to-js-should-follow-to-wasm-call-frame-convention-2.js.default, wasm.yaml/wasm/stress/jitless-wasm-to-js-should-follow-to-wasm-call-frame-convention.js.default (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40254 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38623 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/JavaScriptCore/inspector/agents/InspectorConsoleAgent.h:
```diff
@@ -90,8 +90,8 @@ class JS_EXPORT_PRIVATE InspectorConsoleAgent : public InspectorAgentBase, publi
-    std::unique_ptr<ConsoleFrontendDispatcher> m_frontendDispatcher;
-    RefPtr<ConsoleBackendDispatcher> m_backendDispatcher;
+    const UniqueRef<ConsoleFrontendDispatcher> m_frontendDispatcher;
+    const RefPtr<ConsoleBackendDispatcher> m_backendDispatcher;
```
```suggestion
    const Ref<ConsoleBackendDispatcher> m_backendDispatcher;
```

youennf commented on Source/JavaScriptCore/inspector/agents/InspectorDebuggerAgent.h:
```diff
@@ -246,8 +246,8 @@ class JS_EXPORT_PRIVATE InspectorDebuggerAgent
-    std::unique_ptr<DebuggerFrontendDispatcher> m_frontendDispatcher;
-    RefPtr<DebuggerBackendDispatcher> m_backendDispatcher;
+    const UniqueRef<DebuggerFrontendDispatcher> m_frontendDispatcher;
+    const RefPtr<DebuggerBackendDispatcher> m_backendDispatcher;
```
```suggestion
    const Ref<DebuggerBackendDispatcher> m_backendDispatcher;
```

youennf commented on Source/JavaScriptCore/inspector/agents/InspectorHeapAgent.h:
```diff
@@ -80,8 +80,8 @@ class JS_EXPORT_PRIVATE InspectorHeapAgent : public InspectorAgentBase, public H
-    std::unique_ptr<HeapFrontendDispatcher> m_frontendDispatcher;
-    RefPtr<HeapBackendDispatcher> m_backendDispatcher;
+    const UniqueRef<HeapFrontendDispatcher> m_frontendDispatcher;
+    const RefPtr<HeapBackendDispatcher> m_backendDispatcher;
```
```suggestion
    const Ref<HeapBackendDispatcher> m_backendDispatcher;
```

youennf commented on Source/JavaScriptCore/inspector/agents/InspectorScriptProfilerAgent.h:
```diff
@@ -63,8 +63,8 @@ class JS_EXPORT_PRIVATE InspectorScriptProfilerAgent final : public InspectorAge
-    std::unique_ptr<ScriptProfilerFrontendDispatcher> m_frontendDispatcher;
-    RefPtr<ScriptProfilerBackendDispatcher> m_backendDispatcher;
+    const UniqueRef<ScriptProfilerFrontendDispatcher> m_frontendDispatcher;
+    const RefPtr<ScriptProfilerBackendDispatcher> m_backendDispatcher;
```
```suggestion
    const Ref<ScriptProfilerBackendDispatcher> m_backendDispatcher;
```

youennf commented on Source/JavaScriptCore/inspector/agents/JSGlobalObjectRuntimeAgent.h:
```diff
@@ -50,8 +50,8 @@ class JSGlobalObjectRuntimeAgent final : public InspectorRuntimeAgent {
-    std::unique_ptr<RuntimeFrontendDispatcher> m_frontendDispatcher;
-    RefPtr<RuntimeBackendDispatcher> m_backendDispatcher;
+    const UniqueRef<RuntimeFrontendDispatcher> m_frontendDispatcher;
+    const RefPtr<RuntimeBackendDispatcher> m_backendDispatcher;
```
```suggestion
    const Ref<RuntimeBackendDispatcher> m_backendDispatcher;
```

youennf commented on Source/WebKit/UIProcess/Inspector/WebPageInspectorController.cpp:
```diff
@@ -96,7 +96,7 @@ void WebPageInspectorController::connectFrontend(Inspector::FrontendChannel& fro
 
     if (connectingFirstFrontend)
-        m_agents.didCreateFrontendAndBackend(&m_frontendRouter.get(), &m_backendDispatcher.get());
+        m_agents.didCreateFrontendAndBackend(m_frontendRouter.ptr(), m_backendDispatcher.ptr());
```
It seems `didCreateFrontendAndBackend` implementations do not use these two parameters, maybe we could remove them (not in this patch though).

annevk replied:
Nice find! https://github.com/WebKit/WebKit/pull/47654

annevk committed:
Adopt UniqueRef in JavaScriptCore/inspector
https://bugs.webkit.org/show_bug.cgi?id=295452

Reviewed by Youenn Fablet.

Also mark some additional member variables const and make inspector
code more consistent throughout.

Helps with https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines

Canonical link: https://commits.webkit.org/297041@main
